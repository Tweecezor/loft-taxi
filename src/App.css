*{
    margin:0;
    padding:0;
    box-sizing: border-box;
}
ul{
    list-style: none;
}
a{
    text-decoration: none;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}
.container{
    width:95%;
    margin:0 auto;
}
/* .container--map{
    position: relative;
} */
.header-wrap{
    box-shadow: 0 1px 4px 1px rgba(0, 0, 0, 0.11);
    border-bottom: 1px solid rgba(0,0,0,0.11);
    position: absolute;
    /* top: 0; */
    /* left: 0; */
    background: white;
    width: 100%;
    z-index:10;
}
.map{
    height:100vh;
    position: relative;
}

/* .background::before{
    content:'';
    position: absolute;
    width:100%;
    height: 100%;
    background: rgba(0,0,0,0.3);
    z-index:-1;
} */
.root-wrapper{
    position: relative;
}
input{
    border:none;
    border-bottom: 1px solid #eaeaea;
    padding-bottom: 10px;
    margin-bottom: 10px;
}
.bigTEXT{
    width:100px;
    height: 200px;
    background:white;
    border:2px solid black;
    z-index: 20;
}
.warninng-block{
    position: absolute;
    top:25%;
    left:10%;
    background: #fff;
    padding:40px 40px;
    z-index: 10;
    width:350px;
}
.link-btn{
    padding: 15px 20px;
    background-color: #ffc617;
    border-radius: 4px;
    text-transform: uppercase;
    font-weight: 800;
    border: none;
    margin-bottom: 0;
}
.warning_header{
    margin-bottom: 20px;
}
/* .registration__logo{
    margin-right: 50px;
} */
/* #map {
    position:absolute; top:0; bottom:0; width:100%;
} */

.container-map{
    height: 100%;
    position: relative;
}
#root{
    position: relative;
    overflow: hidden;
  
}
.tooltipError{
    position: absolute;
    line-height: 50px;
    background-color: #ffc617;
    border-radius: 3px;
    bottom: -10%;
    color: black;
    text-align: center;
    left: 50%;
    transform: translateX(-50%);
    padding: 10px;
}
.tooltipErrorAnimation{
    animation: errorAnimation 5s;
}
@keyframes errorAnimation {
    0%{
       bottom: -10%;
    }
    25%{
        bottom:0%;
    }
    50%{
        bottom:0%;
    }
    75%{
        bottom:-10%;
    }
    100%{
        bottom:-10%;
    }

}