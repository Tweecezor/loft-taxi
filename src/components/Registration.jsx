import React,{useState} from 'react';
import '../styles/Registration.css';
import PropTypes from 'prop-types';
import {Link,Redirect} from 'react-router-dom';
import { connect } from 'react-redux';
import {userRegistration,sendRegistrationRequest} from '../Actions/actions'

// class Registration extends React.Component{
const RegistrationComponent = (props) => {

    var[name,setName] = useState('');
    var[surname,setSurname] = useState('');
    var[password,setPassword] = useState('');
    var[email,setEmail] = useState('');


    const handleInputPassword = (e) => {
        setPassword(password = e.target.value);
    }
    const handleInputName = (e) => {
        setName(name = e.target.value);
    }
    const handleInputSurename = (e) => {
        setSurname(surname = e.target.value);
    }
    const handleInputemail = (e) => {
        setEmail(email = e.target.value);
    }

    const showAll = (e)=>{
        e.preventDefault();
        // props.registration(email,name,surename,password);
        props.regOnServer(email,password,name,surname);
    }

        return (
            <div className = 'reg'>
                <div className = 'reg__caption-wrap'>
                    <h2 className = 'reg__caption'>Регистрация</h2>
                </div>
                <div className = 'reg__question'>
                    <span className = 'reg__question-text'>Уже зарегистрирован?</span>
                    <Link to = {'/login'}>Войти</Link>
                </div>
                <form action="" className = 'reg__form' onSubmit = {showAll}>
                    <div className = 'reg__form-wrap reg__form-wrap--email'> 
                        <label htmlFor="email">Адресс электронной почты</label>
                        <input type="email" className = 'reg__form-input' id = 'email' value = {email} onChange = {handleInputemail} />
                    </div>
                    <div className = 'reg__form-wrap reg__form-wrap--name'>
                        <label htmlFor="name">Имя</label>
                        <input type="text" className = 'reg__form-input' id = 'name' value = {name} onChange = {handleInputName} />
                    </div>
                    <div className = 'reg__form-wrap reg__form-wrap--surename'>
                        <label htmlFor="surename">Фамилия</label>
                        <input type="text" className = 'reg__form-input' id='surename' value = {surname} onChange = {handleInputSurename}/>
                    </div>
                    <div className = 'reg__form-wrap reg__form-wrap--password'> 
                        <label htmlFor="password">Пароль</label>
                        <input type="password" className = 'reg__form-input' id = 'password' value = {password} onChange = {handleInputPassword} />
                    </div>
                    <input type="submit" value = 'Зарегистрироваться' className = 'reg__form-submit' />
                </form>
            </div>
        )
}
RegistrationComponent.propTypes = {
    setBlock: PropTypes.func
}
const mapStateToProps = (state) => {
    console.log(state.RegistrationReducer);
    return({})
}

const mapDispatchToProps = (dispatch) => {
    console.log(dispatch)
    return ({
        // registration : (email,name,surename,password)=> {
        //     dispatch(userRegistration(email,name,surename,password));
        // },
        regOnServer : (email,password,name,surname)=> {
            dispatch(sendRegistrationRequest({email,password,name,surname}))
        }
    })
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(RegistrationComponent);