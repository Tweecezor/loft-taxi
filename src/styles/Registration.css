.reg{
    background: #fff;
    width:450px;
    padding: 90px 35px;
    border-radius: 3px;
}
.reg__caption-wrap{
    margin-bottom: 10px;
}
.reg__question{
    margin-bottom: 30px;
}
.reg__form{
    display: flex;

    flex-wrap: wrap;
    justify-content: flex-end;
}
.reg__form-wrap{
    width:100%;
    display: flex;
    flex-direction: column;
}
.reg__form-input{
    border:none;
    border-bottom: 1px solid #eaeaea;
    padding-bottom: 10px;
    margin-bottom: 10px;
}
.reg__form-input:last-child{
    margin-bottom: 30px;
}
.reg__form-wrap--name,
.reg__form-wrap--surename{
    width:47%
}
.reg__form-wrap--name{
    margin-right: 6%;  
}
.reg__form-submit{
    padding: 15px 20px;
    background-color: #ffc617;
    border-radius: 4px;
    text-transform: uppercase;
    font-weight: 800;
    border:none;
}