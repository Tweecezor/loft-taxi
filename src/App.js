import React,{useState} from 'react';

import './App.css';
import Header from './components/Header';

import Map from './components/Map';

import Profile from './components/Profile';
import Background from './components/Background';

import myContext from '../src/context' ;


var currentBlock = 'login';
const {Provider,Consumer} = myContext;






const PAGES = {
    map : () => (    
        <Map />
        ) ,
    profile : () => (
        <section className = 'background'>
            <Profile />
        </section>
    ),
    // welcome : (setPage) => (
       
    //         <Background setPageProp = {setPage} />
      
    // )
}

function App() {

    const login = (email, password) => {
        if (email === 'test' && password === '123') {
            setIsLoggedIn(true)
            // setPage('profile')
        }
    }
    const logout = () => {
        setIsLoggedIn(false)
        // setPage('login')
    }
   
    const [isLoggedIn,setIsLoggedIn] = React.useState(false);
    const [page,setPage] = React.useState('map');
    return (
        <Provider value = { {isLoggedIn,setIsLoggedIn,login,logout}  }>
            {isLoggedIn &&(
                 <div className = 'root-wrapper'>
                    <Header setPageProp = {setPage} />
                    {PAGES[page](setPage)}
                </div>
            )}
            {!isLoggedIn && (
                <Background setPageProp = {setPage} />
            )}
          
           
           
        </Provider>
    );


}

export default App;






// пример как работает код выше
// class newclass extends React.Component {

//     state = {
//         count : 1 
//     }
//     handleClick = ()=>{
//         this.setState({count: this.state.count++})
//     }

//     render(){
//         return(
//             <CompMy onClick={this.handleClick}/>        )
//     }
// }
