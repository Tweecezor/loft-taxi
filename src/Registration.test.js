import React,{useState} from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow,mount} from 'enzyme';
import Registration from './components/Registration';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';


it('Registration Snapshot', () => {
    const tree = renderer.create(<Registration/>).toJSON();
    expect(tree).toMatchSnapshot();
})