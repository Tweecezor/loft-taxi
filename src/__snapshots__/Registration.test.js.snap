// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Snapshot 1`] = `
<div
  className="reg"
>
  <div
    className="reg__caption-wrap"
  >
    <h2
      className="reg__caption"
    >
      Регистрация
    </h2>
  </div>
  <div
    className="reg__question"
  >
    <span
      className="reg__question-text"
    >
      Уже зарегистрирован?
    </span>
    <a
      className="reg__question-link"
      href="#"
      onClick={[Function]}
    >
      Войти
    </a>
  </div>
  <form
    action=""
    className="reg__form"
  >
    <div
      className="reg__form-wrap reg__form-wrap--email"
    >
      <label
        htmlFor="email"
      >
        Адресс электронной почты
      </label>
      <input
        className="reg__form-input"
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="reg__form-wrap reg__form-wrap--name"
    >
      <label
        htmlFor="name"
      >
        Имя
      </label>
      <input
        className="reg__form-input"
        id="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="reg__form-wrap reg__form-wrap--surename"
    >
      <label
        htmlFor="surename"
      >
        Фамилия
      </label>
      <input
        className="reg__form-input"
        id="surename"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="reg__form-wrap reg__form-wrap--password"
    >
      <label
        htmlFor="password"
      >
        Пароль
      </label>
      <input
        className="reg__form-input"
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <input
      className="reg__form-submit"
      type="submit"
      value="Зарегистрироваться"
    />
  </form>
</div>
`;
